app/models/user.rb

```diff
@@ -1,5 +1,11 @@
 class User < ApplicationRecord
   has_secure_password
-  validates_presence_of :firstname, :lastname, :employee_id, :date_of_birth, :joining_day, :designation, :password, message: ->(_, _) { I18n.t('validations.presence') }
+  enum role: { user: 0, admin: 1 }
+  has_one_attached :profile_photo
```

Comment:
maybe you can validate the type of the profile photo such as only allow png, jpeg, jpg files only